### PR TITLE
`Exam mode`: Improve wording for grades with bonus

### PR DESCRIPTION
**Pull request by akesfeden**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added ~multiple~ screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

During a testing session for https://github.com/ls1intum/Artemis/pull/5668, other developers mentioned that the explanations for exam results can be confusing for the students. This PR aims to make the distinction between achieved, max and bonus points clearer. Closes https://github.com/ls1intum/Artemis/issues/5188.

### Description

Added a footer row to display the total values for "Your Points", "Achieved normal points", "Achieved bonus points" columns. Improved explanatory text for exam grade and bonus. Removed redundant whitespace in the exam results mentioned in https://github.com/ls1intum/Artemis/issues/5188.

### Steps for Testing

Prerequisites:
- 1 graded Exam without any bonus exercises or bonus configuration
- 1 graded Exam with bonus exercise(s) but without bonus course/exam
- 1 graded Exam with bonus exercise(s) and bonus course/exam

1. Log in to Artemis

For each prerequisite exam above:

2. Go to the student view of exam summary either with an instructor or a student.
3. Check that the "Your Points", "Achieved normal points", "Achieved bonus points" columns are summed correctly in the "Total" footer row in the summary table at the top.
4. Check the explanatory texts about grade and points (the one starting with "You achieved...") seems correct and clear.

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| exam-points-summary.component.ts | 79.16% | 100% |

### Screenshots

image

### CODE REVIEW

File: src/main/webapp/i18n/en/exam.json [hunk 1]
```diff
@@ -51,15 +51,16 @@
                 "points": {
                     "exercise": "Exercise",
                     "overview": "Overview",
+                    "total": "Total",
                     "yourPoints": "Your Points",
-                    "maxPoints": "Points",
-                    "bonus": "Bonus Points",
-                    "youAchievedWithBonus": "You achieved {{achievedPoints}} out of {{normalPoints}} possible points and an additional {{bonusPoints}} bonus points",
+                    "maxPoints": "Achievable normal points",
```
Comment (simon-hng) [inline src/main/webapp/i18n/en/exam.json:56]:
```suggestion
                    "maxPoints": "Achievable points",
```

File: src/main/webapp/i18n/en/exam.json [hunk 2]
```diff
@@ -51,15 +51,16 @@
...
+                    "maxBonus": "Achievable bonus points",
+                    "youAchievedWithBonus": "You achieved {{achievedPoints}} out of {{maxPoints}} possible points (including bonus)",
                     "youAchieved": "You achieved {{achievedPoints}} out of {{normalPoints}} possible points",
                     "youAchievedFromBonus": {
                         "GRADES_CONTINUOUS": "You got a grade bonus of {{achievedBonus}} from {{bonusFromTitle}}",
                         "GRADES_DISCRETE": "You got a grade bonus of {{achievedBonus}} step(s) according to the grading scale which equals {{gradePointDiff}} from {{bonusFromTitle}}",
-                        "POINTS": "You got {{achievedBonus}} bonus points from {{bonusFromTitle}}"
+                        "POINTS": "You got additional {{achievedBonus}} bonus points from {{bonusFromTitle}}"
```
Comment (simon-hng) [inline src/main/webapp/i18n/en/exam.json:63]:
```suggestion
                        "POINTS": "You got {{achievedBonus}} additional bonus points from {{bonusFromTitle}}"
```

**Review by simon-hng — COMMENTED** [review]
Wording changes look good

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
> You achieved 141 out of 156 possible points (including bonus)

Which number is including the bonus? The 141 or 156?
> Your final points are 151 after the bonus

It seams that it was not the 141. Are you only referring to the 156? This is not really clear to me

File: src/main/webapp/i18n/de/exam.json [hunk 3]
```diff
@@ -51,10 +51,11 @@
...
                     "exercise": "Aufgabe",
                     "overview": "Übersicht",
+                    "total": "Gesamt",
                     "yourPoints": "Deine Punkte",
-                    "maxPoints": "Punkte",
-                    "bonus": "Bonuspunkte",
-                    "youAchievedWithBonus": "Du hast {{achievedPoints}} Punkte von {{normalPoints}} möglichen Punkten, und {{bonusPoints}} Bonuspunkten erreicht.",
+                    "maxPoints": "Erreichbare Punkte",
+                    "maxBonus": "Erreichbare Bonuspunkte",
+                    "youAchievedWithBonus": "Du hast {{achievedPoints}} Punkte von möglichen {{maxPoints}} Punkten (einschließlich Bonus)",
```
Comment (ge65cer) [inline src/main/webapp/i18n/de/exam.json:58]:
```suggestion
                    "youAchievedWithBonus": "Du hast {{achievedPoints}} Punkte von {{maxPoints}} möglichen Punkten (einschließlich Bonuspunkten) erreicht.",
```

File: src/main/webapp/i18n/en/exam.json (hunk 1 above)
Comment (ge65cer) [inline src/main/webapp/i18n/en/exam.json:56]:
```suggestion
                    "maxPoints": "Achievable Points",
```

File: src/main/webapp/i18n/en/exam.json [hunk 4]
```diff
@@ -51,15 +51,16 @@
                 "points": {
                     "exercise": "Exercise",
                     "overview": "Overview",
+                    "total": "Total",
                     "yourPoints": "Your Points",
-                    "maxPoints": "Points",
-                    "bonus": "Bonus Points",
-                    "youAchievedWithBonus": "You achieved {{achievedPoints}} out of {{normalPoints}} possible points and an additional {{bonusPoints}} bonus points",
+                    "maxPoints": "Achievable normal points",
+                    "maxBonus": "Achievable bonus points",
```
Comment (ge65cer) [inline src/main/webapp/i18n/en/exam.json:57]:
```suggestion
                    "maxBonus": "Achievable Bonus Points",
```

File: src/main/webapp/i18n/en/exam.json [hunk 5]
```diff
@@ -51,15 +51,16 @@
...
+                    "youAchievedWithBonus": "You achieved {{achievedPoints}} out of {{maxPoints}} possible points (including bonus)",
```
Comment (ge65cer) [inline src/main/webapp/i18n/en/exam.json:58]:
```suggestion
                    "youAchievedWithBonus": "You achieved {{achievedPoints}} out of a maximum {{maxPoints}} possible points (including bonus points).",
```

File: src/main/webapp/i18n/en/exam.json (hunk 2 above)
Comment (ge65cer) [inline src/main/webapp/i18n/en/exam.json:63]:
```suggestion
                        "POINTS": "You received {{achievedBonus}} additional bonus points from {{bonusFromTitle}}."
```

**Review by ge65cer — CHANGES_REQUESTED** [review]
I personally would not add the bonus points to the maximum points.
So, if I get 100% in all exercise in an exam it would show:
`You achieved 60 out of a maximum 50 possible points.`
Or better:
`You achieved 50 (plus 10 bonus points) out of a maximum 50 possible points.`

**Comment by akesfeden** [comment]
> Get well soon Ata

Thank you very much @simon-hng!

**Comment by akesfeden** [comment]
Reply to @JohannesStoehr 
> > You achieved 141 out of 156 possible points (including bonus)
> 
> Which number is including the bonus? The 141 or 156?

In short, both of them includes it. 156 definitely includes bonus since there are bonus points defined, 141 may also include the bonus depending on whether the student obtained 141 from "normal" or "bonus" points, which we don't explicitly check.
> > Your final points are 151 after the bonus
> 
> It seams that it was not the 141. Are you only referring to the 156? This is not really clear to me

I think the confusing part is there are two kinds of bonus: bonus points from the exercises of the current exam, and additional bonus from another course (EIST in the current example). Both of them are bonus and they both aim to increase the student's final grade but they have different origins.

**Comment by akesfeden** [comment]
> `You achieved 50 (plus 10 bonus points) out of a maximum 50 possible points.`

@ge65cer This one may be a little bit tricky if the student didn't get full bonus points, (e.g. they may got 40 "normal" and 5 "bonus" points or 45 "normal" and no bonus points both resulting in 45 total points.

> `You achieved 60 out of a maximum 50 possible points.`

For the reason above, I think this one may be better, I'll commit changes to write it like this.

File: src/main/webapp/i18n/en/exam.json (hunk 2 above)
Comment (akesfeden) [reply src/main/webapp/i18n/en/exam.json]: Applied manually due to conflicts.

**Comment by nikirovast** [comment]
Hey! The screenshot is still confusing for me. I saw the chat about it, but could you please attach the final version, so it will be more clear?

**Review by canberkanar — APPROVED** [review]
The changes look good to me. 🚀  (I have not reviewed the German translations. )

**Pull request merged**